internal/ir/xds.go

```diff
@@ -2519,6 +2519,10 @@ type Wasm struct {
 	// original URL(either an HTTP URL or an OCI image) and serves it through the
 	// local HTTP server.
 	Code *HTTPWasmCode `json:"httpWasmCode,omitempty"`
+
+	// HostEnvKeys is a list of keys for environment variables from the host envoy process
+	// that should be passed into the Wasm VM.
+	HostEnvKeys []string `json:"hostEnvKeys,omitempty"`
```

Comment:
`hostEnvKeys` is pretty self explanatory, also brainstorming some more options
* `vmConfig.envKeys`
* `env.Keys`
* `envKeys`

Reply:
also linking some past work here from istio https://istio.io/latest/docs/reference/config/proxy_extensions/wasm-plugin/#VmConfig

Reply:
+1 adding a `vmConfig` wrapper for more future proof as other VM settings may be required in the future.
Should we also consider allowing [explicitly specified key-value pairs](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/wasm/v3/wasm.proto#envoy-v3-api-msg-extensions-wasm-v3-environmentvariables).

```
vmConfig:
   env:
      hostEnvKey:
        - host_key_1
        - host_key_2
      cutomEnv:
        - key:  custom_key_1
           value: custom_value_1 
         - key:  custom_key_2
           value: custom_value_2    
```

Reply:
hey @zhaohuabing what value does customEnvs add that cannot be represented via `config` ?

Reply:
Yeah, customEnvs just gives users more options for passing configurations into the Wasm module.

Reply:
`hostEnvKeys` is the name that envoy used for this and I figured it would make it easier for someone to join the dots from the envoy config if it matched here.

For a vmConfig wrapper, would this just contain the `hostEnvKeys` or the `config` block too? happy to add it, but is there value to the extra nesting?

Reply:
yeah I'm a +1 for `hostEnvKeys`, it leaves room for `customEnvs` . The other way to frame this is `env.HostKeys`, which leaves room for `env.Custom`.